src/node_file.cc

```diff
@@ -2364,17 +2364,17 @@ static void Read(const FunctionCallbackInfo<Value>& args) {
   char* buf = buffer_data + off;
   uv_buf_t uvbuf = uv_buf_init(buf, len);
 
-  FSReqBase* req_wrap_async = GetReqWrap(args, 5);
-  if (req_wrap_async != nullptr) {  // read(fd, buffer, offset, len, pos, req)
+  if (argc > 5) {  // read(fd, buffer, offset, len, pos, req)
+    FSReqBase* req_wrap_async = GetReqWrap(args, 5);
+    CHECK_NOT_NULL(req_wrap_async);
     FS_ASYNC_TRACE_BEGIN0(UV_FS_READ, req_wrap_async)
     AsyncCall(env, req_wrap_async, args, "read", UTF8, AfterInteger,
               uv_fs_read, fd, &uvbuf, 1, pos);
-  } else {  // read(fd, buffer, offset, len, pos, undefined, ctx)
-    CHECK_EQ(argc, 7);
-    FSReqWrapSync req_wrap_sync;
+  } else {  // read(fd, buffer, offset, len, pos)
+    FSReqWrapSync req_wrap_sync("read");
     FS_SYNC_TRACE_BEGIN(read);
-    const int bytesRead = SyncCall(env, args[6], &req_wrap_sync, "read",
-                                   uv_fs_read, fd, &uvbuf, 1, pos);
+    const int bytesRead = SyncCallAndThrowOnError(
```

Comment:
you need to return nothing if bytesRead is a libuv error.

Reply:
Thank for your comments.
I added it.Thank for your comments.
I added it.